Xamarin.Forms.Core/Picker.cs

```
@@ -7,246 +7,249 @@
+
+        public static readonly BindableProperty TitleProperty = BindableProperty.Create(nameof(Title), typeof(string), typeof(Picker), default(string));
+
+        public static readonly BindableProperty SelectedIndexProperty = BindableProperty.Create(nameof(SelectedIndex), typeof(int), typeof(Picker), -1, BindingMode.TwoWay,
+            propertyChanged: OnSelectedIndexChanged,
+            coerceValue: CoerceSelectedIndex);
+
+        public static readonly BindableProperty SelectedValueMemberPathProperty = BindableProperty.Create(nameof(SelectedValueMemberPath), typeof(string), typeof(Picker));
+
+        public static readonly BindableProperty ItemsSourceProperty = BindableProperty.Create(nameof(ItemsSource), typeof(IList), typeof(Picker), default(IList), propertyChanged: OnItemsSourceChanged);
+
+        public static readonly BindableProperty DisplayMemberPathProperty = BindableProperty.Create(nameof(DisplayMemberPath), typeof(string), typeof(Picker), propertyChanged: OnDisplayMemberPathChanged);
+
+        public static readonly BindableProperty SelectedItemProperty = BindableProperty.Create(nameof(SelectedItem), typeof(object), typeof(Picker), null, BindingMode.TwoWay, propertyChanged: OnSelectedItemChanged);
+
+        public static readonly BindableProperty DisplayFuncProperty =
+            BindableProperty.Create(
+                nameof(DisplayFunc),
+                typeof(Func<object, string>),
+                typeof(Picker));
+        public Picker()
+        {
+            ((ObservableList<string>)Items).CollectionChanged += OnItemsCollectionChanged;
+        }
+
+        public string DisplayMemberPath
+        {
+            get { return (string)GetValue(DisplayMemberPathProperty); }
+            set { SetValue(DisplayMemberPathProperty, value); }
+        }
+
+        public Func<object, string> DisplayFunc
```

Perhaps that would be better. If the converter doesn't return `string` then we could throw an exception. Perhaps include both options?